third_party/CMakeLists.txt

```diff
@@ -93,3 +93,45 @@ if(VELOX_ENABLE_ARROW)
                                            ${ARROW_LIBDIR}/libparquet.a)
 
 endif()
+
+if(VELOX_ENABLE_QPL)
+  message(STATUS "Building QPL from source")
+  set(QPL_PREFIX "${CMAKE_CURRENT_BINARY_DIR}/qpl_ep/install")
+  set(QPL_STATIC_LIB_NAME
+      ${CMAKE_STATIC_LIBRARY_PREFIX}qpl${CMAKE_STATIC_LIBRARY_SUFFIX})
+  set(QPL_STATIC_LIB "${QPL_PREFIX}/lib/${QPL_STATIC_LIB_NAME}")
+
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -ldl -laccel-config -L/usr/lib64")
+
+  set(QPL_CMAKE_ARGS
+      -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}
+      -DCMAKE_INSTALL_LIBDIR=${QPL_PREFIX}/lib
+      -DCMAKE_INSTALL_PREFIX=${QPL_PREFIX}
+      -DCMAKE_CXX_FLAGS=${CMAKE_CXX_FLAGS}
+      -DEFFICIENT_WAIT=OFF
+      -DQPL_BUILD_TESTS=OFF
+      -DCMAKE_C_COMPILER=gcc
+      -DCMAKE_CXX_COMPILER=g++
+      -DQPL_BUILD_EXAMPLES=OFF
+      -DQPL_LIB=ON)
+
+  ExternalProject_Add(
+    qpl_ep
+    ${EP_LOG_OPTIONS}
+    URL https://github.com/intel/qpl/archive/refs/tags/v1.3.0.tar.gz
+    BUILD_BYPRODUCTS "${QPL_STATIC_LIB}"
+    CMAKE_ARGS ${QPL_CMAKE_ARGS})
+
+  file(MAKE_DIRECTORY "${QPL_PREFIX}/include")
+
+  add_library(qpl::qpl STATIC IMPORTED GLOBAL)
+  set(QPL_LIBRARIES ${QPL_STATIC_LIB})
+  set(QPL_INCLUDE_DIRS "${QPL_PREFIX}/include")
+  target_link_libraries(qpl::qpl INTERFACE /usr/lib64/libaccel-config.so)
```

Comment:
Hardcoding paths like this is a total no go, please use the variables that cmake provides for ep (or better yet as mentioned above use FC and skip all of this ;) )